astropy/io/fits/hdu/hdulist.py

```diff
@@ -521,12 +521,7 @@ def index_of(self, key):
         nfound = 0
         found = None
         for idx, hdu in enumerate(self):
-            name = hdu.name
-            if isinstance(name, string_types):
-                name = name.strip().upper()
-            # 'PRIMARY' should always work as a reference to the first HDU
-            if ((name == _key or (_key == 'PRIMARY' and idx == 0)) and
-                (_ver is None or _ver == hdu.ver)):
+            if self._ext_equal((_key, _ver), hdu, idx):
```

Comment:
yes I tried to avoid as much of it as I could think of, but its either have two functions to 90% the same or some duplicate work.
The extra work is negligible in terms of cpu time so I opted for less code